src/main/java/org/mandas/docker/client/DefaultDockerClient.java

```diff
@@ -264,7 +259,7 @@ private LoadProgressHandler(ProgressHandler delegate) {
     }
 
     private Set<String> getImageNames() {
-      return ImmutableSet.copyOf(imageNames);
+      return unmodifiableSet(new HashSet<>(imageNames));
```

Comment:
Should this be typed?
I'm afraid all of these "should this be typed?" comments are a waste, but I'll keep calling out in case you want to type them.